### PR TITLE
Refs #26729 - Feat/switch api get

### PR DESCRIPTION
**Pull request by MariaAga**
blocked by https://github.com/theforeman/foreman/pull/7069

### CODE REVIEW

File: webpack/assets/javascripts/react_app/API.js [hunk 1]
```diff
@@ -1,40 +1,9 @@
-import axios from 'axios';
-import './APITestSetup';
+import { API } from './redux/API';
+import { deprecate } from './common/DeprecationService';
 
-const getcsrfToken = () => {
-  const token = document.querySelector('meta[name="csrf-token"]');
+deprecate(
```
Comment (ekohl) [inline webpack/assets/javascripts/react_app/API.js]: Note we dropped the version parameter default so you need to be explicit.

**Comment by ares** [comment]
I'm sorry, this now needs again, as we realized during PR triage

**Comment by ezr-ondrej** [comment]
@MariaAga the rebase didn't solve the issue it seems :thinking:

**Comment by Ron-Lavi** [comment]
@MariaAga can you please rebase?

**Comment by sharvit** [comment]
@MariaAga what is the status here?

**Comment by ares** [comment]
Needs rebase now

**Comment by Ron-Lavi** [comment]
@MariaAga would it make sense to wait with this PR since it is conflicting with the way we decided to go in #7222 ?

**Comment by MariaAga** [comment]
Yes I will wait until your PR is merged so it will use the new implementation

**Comment by tbrisker** [comment]
@MariaAga looks like the other PR was merged, do you plan on continuing with this or should we close it?

**Comment by MariaAga** [comment]
I'll close it for now. too many changes in the api middleware

**Pull request closed without merging**